### DISCUSSION TITLE
Use nm-applet and blueberry-tray and avoid full plasma installation

### DISCUSSION THREAD

**Discussion by Nevrdid** [end-4/dots-hyprland#341] [Ideas]
Hi, i would like to purpose to use `nm-applet --indicator` and `blueberry-tray` 
Here is a patch to fix blueberry-tray for wayland : 
[blueberry-wayland.patch.txt](https://github.com/end-4/dots-hyprland/files/14665355/blueberry-wayland.patch.txt)
( It's a updated version of patch used for blueberry-wayland aur package and require https://archlinux.org/packages/extra/x86_64/libayatana-appindicator/)
instead of rely on ags half-made applets. Thanks x)
Also, as i'm here, i would like to avoid installing all the plasma suite just to have useless plasma-browser-integration ( the mpris work well without it for me)

**Comment by end-4** [comment]
> purpose to use nm-applet --indicator and blueberry-tray

- If "purpose" you mean "propose", sorry, nope. I'd rather not have ugly tray items and would rather use an app. 
- But of course, feel free to use them yourself by adding `exec-once`s for them in `~/.config/hypr/custom/execs.conf`

> the mpris work well without it for me

- :eyes:... does it work with browser? like properly provide cover art? because I've tried many times and it doesn't.
  - If yes, what browser are you using?

**Reply by Nevrdid** [reply]
My bad for the wrong word use, and thanks to noticed it to me x')
Well is like nearly the sames ones and i'm pretty sure we could change icons anyway. I personally prefer the functionality solution for that ^^'
![image](https://github.com/end-4/dots-hyprland/assets/13476270/eb8116ee-0a57-4e66-a04c-44f284e0a7c3)

> 👀... does it work with browser? because i've tried many times and it doesn't

![image](https://github.com/end-4/dots-hyprland/assets/13476270/c8f28cdd-f93a-429e-be20-48b124215c10)
And buttons works well. Can't we just hide playtime as it is not required imo x')

![image](https://github.com/end-4/dots-hyprland/assets/13476270/2aefbc2b-5aa3-44c9-b56e-f11eb19d7784)

**Reply by end-4** [reply]
About those wifi and bluetooth buttons, I can implement those functionality in ags but only choosing network and bluetooth device, no wifi password entering... I think that isn't too bad

About mpris without plasma integration, interestingly, 
- With my config, it works with Thorium (basically chromium) but not Firefox
- With [Aylur's config](https://github.com/Aylur/dotfiles), mpris integration does work with Firefox. This suggests it's a problem with my config, so I'll see if I can fix it

**Reply by end-4** [reply]
okay that mpris fix was easier than i thought lol. just had to copy that part of aylur's config
f6fa86e26aba99c224173cfed1deafe01d718921

**Reply by Nevrdid** [reply]
well, i commented the org.mpris.MediaPlayer2.firefox line in musiccontrols.js also. It sure not work with it.

**Reply by end-4** [reply]
i still want time indication so i made the thing work nicely whether you have plasma integration or not: 85fc34161781fce6043425a40be910cbfabbf819

**Reply by Nevrdid** [reply]
However, i still think default shouldn't be to install the full plasma suite x') Can't you ask at install if user want it or no ? I also have kde + hypr on my another laptop and i wouldn't uninstall integration on it . If so, just hide the time stuff if it is not defined.

**Reply by end-4** [reply]
@clsty I gotta bother you again
Can you make a step like this in the install script
```
Would you like playtime of media in Firefox to be shown on the music controls widget?
If yes, we're going to install `plasma-browser-integration` (about 250MiB)
```
ps: i saw you saying you're busy, so take your time. no rush or pressure or anything of course

**Reply by Nevrdid** [reply]
hum, i'm pretty sure it's more than 600Mb with all dependencies. But don't remember exactly the amount.
edit : 777Mb
[plasma-browser-integration_install.log](https://github.com/end-4/dots-hyprland/files/14666795/plasma-browser-integration_install.log)

**Reply by end-4** [reply]
I must have something else that requires full installation of plasma stuff then. ¯\_(ツ)_/¯
a `yay -Rs plasma-browser-integration` only strips 249.71 MiB for me

**Reply by Nevrdid** [reply]
Probably qt5ct and qt5-wayland

**Reply by Nevrdid** [reply]
Also, i finally switch to blueman for applet as is allow me to remove deprecated gnome-bluetooth required by blueberry and have no dependency ~~and i installed swaync as notification deamon ( looks it is missing from the config)~~

**Reply by end-4** [reply]
don't use these dotfiles at this point xd
isn't gnome bluetooth required by ags?

**Reply by Nevrdid** [reply]
xD isn't customization allowed ? I mean, i didn't modify that much.
Gnome-bluetooth-3.0 is as optional but the one required by blueberry is another deprecated package.
gnome-bluetooth-3.0 is also the one required by gnome-control-center and anyway just not work if the d.e. isn't gnome.

**Reply by Nevrdid** [reply]
Great :+1: I updated and like the new widgets, gonna close the discussion :)
Also, i looked back at what installed me the full plasma suite and it's well the browser-integration. i guess it didn't report you with -Rs because you installed something else on your system which require some of them but for a clean gnome install, it's 800Mib unrequired.

**Comment by clsty** [comment]
> @clsty I gotta bother you again Can you make a step like this in the install script
> 
> ```
> Would you like playtime of media in Firefox to be shown on the music controls widget?
> If yes, we're going to install `plasma-browser-integration` (about 250MiB)
> ```
> 
> ps: i saw you saying you're busy, so take your time. no rush or pressure or anything of course

I have a better idea on this to try, but I can't do anything until next month.

**Reply by clsty** [reply]
@end-4 Done in https://github.com/end-4/dots-hyprland/commit/9ad16495f42a4ca21cf5886d2d2c1fc0f6ae7f2c
